test/eventpattern.test.ts

```diff
@@ -0,0 +1,1517 @@
+import { ebEventPatternTestCase, ebEventPatternTestCaseError } from "./util";
+import { EventBusRuleInput, EventPredicateFunction } from "../src/event-bridge";
+import { reflect } from "../src";
+
+type TestEvent = EventBusRuleInput<{
```

Comment:
I'm wondering why we need the developer to always use this wrapper? Can't they instead just declare their type and then our constructs wrap it?

Reply:
My initial thought was allowing the developer to provide rich types for any of the fields in the wrapper.

```
type MyEvent = EventBusRuleInput<{
   value: string
}> & {
    version: 1 | 2 | 3
}
```

and to support rich unions

```
type MyEvent = EventBusRuleInput<{ someField: string }, 'customDetailType'>;

new EventBus<StepFunctionEvent | MyEvent>( )
   .when(event => event['detail-type'] === customDetailType && event.detail.someField)
```

Thoughts on how to handle this more gracefully?

Reply:
Let's open an issue and continue the discussion.

Reply:
#94